rootfs/deploy.py

```diff
@@ -26,6 +24,19 @@ def log_output(stream, decode):
         else:
             print(chunk.decode('utf-8'))
 
+
+def stream_error(stream):
+    """Translate docker stream errors into a more digestable format"""
+    error = False
+    for chunk in stream:
+        # error handling requires looking at the response body
+        if 'error' in chunk:
```

Comment:
Guess you technically already have half of it https://github.com/deis/controller/blob/master/rootfs/registry/dockerclient.py#L135-L154